### DISCUSSION TITLE
Installation on M2 Max Desktop

### DISCUSSION THREAD

**Discussion by SnowySnowySnowy** [talmolab/sleap#1688] [Help!] [answered]
I installed mambaforge and then installed SLEAP using conda package. When I try to run  `mamba activate sleap`, I get the following message: 

`Run 'mamba init' to be able to run mamba activate/deactivate
and start a new shell session. Or use conda to activate/deactivate.`

I tried running `mamba init`, 

`==> For changes to take effect, close and re-open your current shell. <==`

`Added mamba to /Users/em47536/.bash_profile`

`==> For changes to take effect, close and re-open your current shell. <==`

However, this doesn't seem to fix the issue, as it just shows me the same messages if I close and re-open my shell. 

As suggested in other forums [https://github.com/talmolab/sleap/discussions/1536](url), I did`open .zshrc`, which shows this: 

`# >>> conda initialize >>>`
`# !! Contents within this block are managed by 'conda init' !!`
`__conda_setup="$('/Users/em47536/mambaforge3/bin/conda' 'shell.zsh' 'hook' 2> /dev/null)"`
`if [ $? -eq 0 ]; then`
 `   eval "$__conda_setup"`
`else`
`    if [ -f "/Users/em47536/mambaforge3/etc/profile.d/conda.sh" ]; then
`        . "/Users/em47536/mambaforge3/etc/profile.d/conda.sh"
`    else
`        export PATH="/Users/em47536/mambaforge3/bin:$PATH"`
`    fi`
`fi`
`unset __conda_setup`
`# <<< conda initialize <<<`

What I am confused about is whether this should be the mamba initialize file, not conda? Should I comment out this whole file? Or am I misunderstanding something?

**Comment by Marielamend** [comment]
Hello @SnowySnowySnowy!

Have you tried to activate your environment with `conda activate sleap` ? 

Best,
Mariela

**Reply by SnowySnowySnowy** [reply]
For some reason I thought I was not supposed to use conda, is it ok to use that rather than mamba? I am not super familiar with either, so my apologies if that's a silly question

**Reply by Marielamend** [reply, marked as answer]
Hello @SnowySnowySnowy,

Not a silly question at all! 
_Some background info:_
Mamba and conda are both package managers but the mamba solver is a lot faster at installing packages than conda. Mamba is a CLI tool to manage conda environments.

To break down this message:

`Run 'mamba init' to be able to run mamba activate/deactivate and start a new shell session. Or use conda to activate/deactivate.`
This is saying Mamba package manager needs to be initialized before you can use its environment activation and deactivation commands (“mamba activate”)

If you prefer not to use “mamba init”, you can use `conda activate sleap` 

As you can see in this discussion post #1536
>If you don't see this block, then something is amiss as the mamba init command should add this block for us. The thing that is likely the issue is that you also have a conda initialize block in you .zshrc file (perhaps left behind from a previous conda installation):

As you showed us in your post, you are seeing ">>> conda initialize >>>" in your .zshrc file, which is most likely why you are having issues running `mamba init` since...
>Having both mamba and conda will cause conflicts as was the case in [this forum](https://forums.fast.ai/t/mamba-activate-not-working-mac-m-chip/105585). 


- Try using command `conda activate sleap`, this should work in activating the environment, but let us know if it does not work!

Best,
Mariela

**Comment by SnowySnowySnowy** [comment]
Oh my goodness. Thank you, that worked!
